### PR TITLE
feat(Commerce Atomic): add atomic-commerce-result-list

### PR DESCRIPTION
**Pull request by fpbrault**
This PR adds the atomic-commerce-result-list component

https://coveord.atlassian.net/browse/KIT-3047

### CODE REVIEW

File: packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  @Prop({reflect: true}) density: 'normal' | 'compact' = 'normal';
+
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize: number = 400;
+
+  public initialize() {
+    if (this.host.innerHTML.includes('<atomic-result-children')) {
+      console.warn(
```
Comment (olamothe) [inline packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx:61]: This is incorrect: There will be no differentiation/special support for folding with atomic for commerce: Folding is always enabled, and done by the commerce API. This warning is superfluous.

File: packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    if (
+      products.length === 0 &&
+      !this.productListingState?.isLoading &&
+      !this.searchState?.isLoading
+    ) {
+      return <div>No products found.</div>;
+    }
+
+    return (
```
Comment (olamothe) [inline packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx:105]: I'm guessing this is all temporary, until we ad support for result templates as usual ? 

If so, I'd prefer a TODO to mention that we need to wire result template system.

File: packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          icon={StarIcon}
+        ></atomic-icon>
+      );
+    }
+
+    return stars;
+  }
+
+  public render() {
+    const products =
```
Comment (olamothe) [inline packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx:92]: There is a bunch of functional components I've added to handle these "correctly". If you look at `atomic-result-list`, you'll see things like `ItemListGuard`, `DisplayWrapper`, `Placeholders` etc.

**Review by olamothe — APPROVED** [review]
As long as it is understood that we need to complete redo the `render()` function as it currently exist (with TODOS), I'm fine with merging this.

File: packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx (hunk 2 above)
Comment (fpbrault) [reply packages/atomic/src/components/commerce/atomic-commerce-result-list/atomic-commerce-result-list.tsx:105]: yep all temporary, I'll add a TODO

**Review by louis-bompart — APPROVED** [review]

**Review by Spuffynism — APPROVED** [review]

**Pull request merged**